cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -41,6 +41,20 @@ endmacro()
 macro (build_addon target prefix libs)
   addon_version(${target} ${prefix})
 
+  # Check addons version is equal or less to Kodi's Version
+  # 1. If addon version higher as Kodi version, seems it not to correct API and
+  #    need to abort, e.g. to prevent on new addon release an upload to wrong
+  #    Kodi addon mirror.
+  # 2. If addon version is lower can it maybe still work and is further usable,
+  #    as there maybe no API changes since them.
+  #    Bring only a warning message in this case.
```

Comment:
```
  # Compare addon version to Kodi's Version
  # 1. If addon version is larger then Kodi version, it's not the correct API and
  #    need to abort to prevent new addon release and upload to wrong
  #    Kodi addon mirror.
  # 2. If addon version is lower than Kodi version, it may or may not work
  #    due to API changes. Show a warning message in this case.
```